src/Validator/Validate.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+
+namespace App\Validator;
+
+use App\Entity\Painting;
+use App\Enum\RequestType;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\Validator\Constraints\Required;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class Validate implements ValidateInterface
+{
+    private $validator;
+    private $entityManager;
+
+    public function __construct(ValidatorInterface $validator, EntityManagerInterface $entityManagerInterface)
+    {
+        $this->validator = $validator;
+        $this->entityManager = $entityManagerInterface;
+    }
+
+    public function validateNewRequest(Request $request, $requestType){
+        $error = $this->requestValidator($request, $requestType);
+        if ($error != null){
+            return $error;
+        }
+
+        if ($requestType === RequestType::$REQUEST_SEARCH){
+            $error = $this->searchQueryValidator($request);
+            if ($error != null){
+                return $error;
+            }
+        } else {
+            $error = $this->pantingValidator($request, $requestType);
+            if ($error != null) {
+                return $error;
+            }
+        }
+        return null;
+    }
+
+    public function requestValidator(Request $request, $type)
+    {
+        switch ($type) {
```

Comment:
it is better to call the validator function directly no need for the case statement here,
imaging if we added new method/type 
why we have to change this ?